### PR TITLE
Versioned objects.

### PR DESCRIPTION
**Pull request by corystone**

### CODE REVIEW

**Comment by corystone** [comment]
Putting this up here before I take off for vacation. There is a bunch of untested error handling. The swift functests will be useful once we get fast POST.

**Comment by gholt** [comment]
Working on this at #172

File: proxyserver/middleware/versioned_writes.go [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+		srv.StandardResponse(writer, 500)
+	}
+	return true
+}
+
+func (v *versionedWrites) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	apiReq, account, container, object := getPathParts(request)
+	if container == "" && object == "" {
+		v.next.ServeHTTP(writer, request)
+	}
```
Comment (gholt) [inline proxyserver/middleware/versioned_writes.go]: @redbo Noticed you missed a return here.

**Comment by gholt** [comment]
I was going to work on getting this merged while you were out in #172 but abandoned that idea for now. By the time we get it sorted, you'll probably be back. :)

**Pull request closed without merging**